src/robot/libraries/String.py

```diff
@@ -587,10 +591,23 @@ def generate_random_string(self, length=8, chars='[LETTERS][NUMBERS]'):
         | ${low} = | Generate Random String | 12 | [LOWER]         |
         | ${bin} = | Generate Random String | 8  | 01              |
         | ${hex} = | Generate Random String | 4  | [NUMBERS]abcdef |
+        | ${rnd} = | Generate Random String | 5-10 | # Generates a string 5 to 10 characters long |
+
+        Giving ``length`` as a range of values is new in Robot Framework 5.0.
         """
         if length == '':
             length = 8
-        length = self._convert_to_integer(length, 'length')
+        if '-' in str(length):
+            try:
+                min_length, max_length = length.split('-')
+                length = randint(self._convert_to_integer(min_length, "length"),
+                                 self._convert_to_integer(max_length, "length"))
+            except ValueError:
+                raise ValueError("Length is not a valid range of integers: {0}".format(length))
+            except AttributeError:
+                pass    # To keep behaviour as it is with negative numbers
```

Comment:
Catching and ignoring AttributeError is too magic even with a comment.

Reply:
Perhaps it would be easier to check that the range is in correct format so that no error checking would be needed here in general. Something like this:
```python
if isinstance(length, str) and re.match(r'^\d+-\d+$', length):
    min_length, max_length = length.split('-')
    ...
else:
    length = ...
```

Reply:
Moved all error handling to `_convert_to_integer` as suggested.